shared/fs/index.js

```diff
@@ -1,30 +1,110 @@
 // @flow
-import * as React from 'react'
-import {globalStyles, globalMargins} from '../styles'
-import {Box, Button, Text, ButtonBar} from '../common-adapters'
-
-type Props = {
-  counter: number,
-  you: ?string,
-  increase: () => void,
-  increase10: () => void,
+import React, {PureComponent} from 'react'
+import {FolderVisibility} from '../constants/types/fs'
+import {globalStyles, globalColors, globalMargins} from '../styles'
+import {isMobile} from '../constants/platform'
+import {Box, Button, ClickableBox, Icon, List, Text} from '../common-adapters'
+import {type IconType} from '../common-adapters/icon'
+
+const stylesCommonCore = {
+  alignItems: 'center',
+  borderBottomColor: globalColors.black_05,
+  borderBottomWidth: 1,
+  justifyContent: 'center',
 }
 
-const Fs = ({counter, you, increase, increase10}: Props) => (
-  <Box style={containerStyle}>
-    <Text type="Header">Hi {you}!</Text>
-    <Text type="BodySemibold">Count: {counter}</Text>
-    <ButtonBar align="flex-start">
-      <Button type="Primary" onClick={increase} label="Up by 1" />
-      <Button type="Primary" onClick={increase10} label="Up by 10" />
-    </ButtonBar>
-  </Box>
-)
+const stylesCommonRow = {
+  ...globalStyles.flexBoxRow,
+  ...stylesCommonCore,
+  minHeight: isMobile ? 64 : 48,
+  padding: 8,
+}
 
-const containerStyle = {
+const stylesContainer = {
   ...globalStyles.flexBoxColumn,
+  ...globalStyles.fullHeight,
   flex: 1,
-  padding: globalMargins.xlarge,
 }
 
+const rootFolders = [
+  'private',
+  'public',
+  'team',
+]
+
+type FolderHeaderProps = {
+  title: string,
+}
+
+type FileRowProps = {
+  path: string,
+  icon: IconType,
+  showFileData: () => void,
+}
+
+type FolderProps = {
+  path: string,
+  visibility: FolderVisibility,
+  items: Array<string>,
+}
+
+const FolderHeader = ({title}: FolderHeaderProps) => (
+  <Box>
+    <Box style={{...stylesCommonRow, alignItems: 'center', borderBottomWidth: 0}}>
```

Comment:
This style is fine and we do it everywhere, just a heads up if you have items that are mutating a bunch (likely not in this case) when this render happens it'll be making this style object over and over and it'll fail any equality comparison so it'll force a re-render. This isn't a problem here as the connect call is pure and will not rerender if nothing is changing, but if we did have more stuff rendering this style could cause render thrashing.

tldr; {foo: 'bar'} !== {foo: 'bar'}

Reply:
So what's the right answer here?

Reply:
just pull it into its own variable. below just make a const FolderBoxStyle = {...stylesCommonRow, ...} etc

Reply:
Does this apply to all the `style={{foo}}` attributes?

Reply:
it applies to all props period

Reply:
We seem to have this issue on the Devices screen then, where we have tons of `style={{foo}}`

Reply:
this is why i mentioned if the things mutate a bunch. on the devices screen they never mutate so it renders once on mount

Reply:
I think this would still cause re-rendering since `style` is a new object every time. Did you intend to do this because `FolderHeader` rarely gets re-rendered, or is this supposed to be `FolderBoxStyle` as well?

Reply:
Yeah the header will never be re-rendered.